software/main.py

```diff
@@ -104,198 +136,232 @@ def main_loop():
                 frame = 0
                 end = time.time()
                 fps = 30/(end-start)
-                start = end
-                #[Object: x=7; y=212; size=15.0; distance=212; exists=True]  
+                start = end 
                 print(fps)  
 
+            
+            try:
+            
+                if ref_cmds == True:
+                    run, blue = ref.get_current_referee_command()
+                    print(run, blue )
+                    if run == True and First_Ref==1:
+                        state = State.FIND_BALL
+                        if blue == True:
+                            basket_color = BasketColor.BLUE
+                        else:
+                            basket_color = BasketColor.MAGENTA
+                        First_Ref = 0
+                        continue
+                    elif run == False and First_Ref==0:
+                        state = State.WAITING
+                        First_Ref = 1
+                elif ref_cmds == False and First_Ref==1:
+                    state = State.FIND_BALL
+                    First_Ref = 0
+        
+            except:
+                print("Server client communication failed.")

```

Comment:
Maybe try to restart the communication logic? When can this error happen? When no connection could be established?